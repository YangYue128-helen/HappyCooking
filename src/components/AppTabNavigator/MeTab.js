import React, {Component} from 'react';
import {
    StyleSheet, Text, View, Image, Dimensions,
} from 'react-native';
import {Icon, Container, Content, Header, Left, Body, Right, Button} from 'native-base';
import EntypoIcon from 'react-native-vector-icons/Entypo'
import CardCompo from "../CardCompo";
import { createStackNavigator } from "@react-navigation/stack";

var images = [
    require('../../images/c1.png'),
    require('../../images/c2.png'),
    require('../../images/c3.png'),
    require('../../images/c4.png'),
    require('../../images/c5.png'),
    require('../../images/c6.png'),
    require('../../images/c7.png'),
    require('../../images/c8.png'),
    require('../../images/c9.png'),
    require('../../images/c10.png'),
]

var {width, height} = Dimensions.get('window')

class MeTab extends Component {
    constructor() {
        super()
        this.state = {
            activeIndex: 0
        }
    }

    segmentClicked = (index) => {
        this.setState({
            activeIndex: index
        })
    }

    renderSectionOne =() => {
        return images.map((image, index) => {
            return(
                <View key={index} style={[{width:(width)/3}, {height: (width)/3},
                    {marginBottom:2},
                    index % 3 !== 0 ? {paddingLeft: 2} : {paddingRight: 0}
                ]}>
                    <Image style={{flex: 1, width: undefined, height: undefined}}
                        source={image}
                    />
            </View>
            )
        })
    }

    renderSection = () => {
        if(this.state.activeIndex == 0) {
            return(
                <View style={{flexDirection:'row', flexWrap:'wrap'}}>
                    {this.renderSectionOne()}
                </View>
            )
        } else if(this.state.activeIndex == 1) {
            return (
                <View>
                    <CardCompo imageSource="1" likes='101 likes'/>
                    <CardCompo imageSource="2" likes='1 likes'/>
                    <CardCompo imageSource="3" likes='505 likes'/>
                    <CardCompo imageSource="4" likes='99 likes'/>
                    <CardCompo imageSource="5" likes='5 likes'/>
                    <CardCompo imageSource="6" likes='76 likes'/>
                    <CardCompo imageSource="7" likes='88 likes'/>
                </View>
            )

        } else if(this.state.activeIndex == 2) {

        }
    }

    render() {
        return(
            <Container style={{flex: 1, backgroundColor: 'white'}}>
                <Header>
                    <Left><Icon name="md-person-add" style={{paddingLeft: 10}}></Icon></Left>
                    <Body><Text>HelenYang</Text></Body>
                    <Right><EntypoIcon name="back-in-time" style={{paddingRight:10, fontSize: 32}}/></Right>
                </Header>
                <Content>
                    <View style={{paddingTop: 10}}>
                        <View style={{flexDirection: 'row'}}>
                            <View style={{flex: 1, alignItems: 'center'}}>
                                <Image source={require('../../images/Me.png')}
                                    style={{width:75, height:75, borderRadius:37.5}}/>
                            </View>
                            <View style={{flex: 3}}>
                                <View style={{flexDirection: 'row', justifyContent: 'space-around'}}>
                                    <View style={{alignItems: 'center'}}>
                                        <Text>20</Text>
                                        <Text style={{fontSize:10, color:'grey'}}>posts</Text>
                                    </View>
                                    <View style={{alignItems: 'center'}}>
                                        <Text>30</Text>
                                        <Text style={{fontSize:10, color:'grey'}}>recipes</Text>
                                    </View>
                                    <View style={{alignItems: 'center'}}>
                                        <Text>50</Text>
                                        <Text style={{fontSize:10, color:'grey'}}>followers</Text>
                                    </View>
                                </View>

                                <View style={{flexDirection:'row', paddingTop:10}}>
                                    <Button bordered dark
                                            style={{flex:3, marginLeft:10, justifyContent:'center', height: 30}}>
                                        <Text>Edit Profile</Text>
                                    </Button>
                                    <Button bordered dark
                                            style={{flex:1, marginLeft:5, marginRight: 10, justifyContent:'center', height: 30}}
                                            onPress = { () => this.props.navigation.navigate("Setting") }>
                                        <Icon name='settings' style={{alignItems: 'center', marginBottom:-5}}/>
                                    </Button>
                                </View>
                            </View>
                        </View>

                        <View style={{paddingVertical:10, paddingHorizontal: 10}}>
                            <Text style={{fontWeight:'bold'}}>Yang Yue</Text>
                            <Text>Chinese cuisine | Korean cuisine</Text>
                        </View>
                    </View>

                    <View>
                        <View style={{flexDirection:'row', justifyContent: 'space-around',
                        borderTopWidth: 1, borderTopColor: '#eeeeee'}}>
                            <Button
                                transparent
                                onPress={() => this.segmentClicked(0)}
                                active={this.state.activeIndex == 0}
                            >
                                <Icon name='ios-keypad'
                                      style={[this.state.activeIndex == 0 ? {} : {color: 'grey'}]}/>
                            </Button>
                            <Button
                                transparent
                                onPress={() => this.segmentClicked(1)}
                                active={this.state.activeIndex == 1}
                            >
                                <Icon name='ios-list'
                                      style={[this.state.activeIndex == 1 ? {} : {color: 'grey'}]}/>
                            </Button>
                            <Button
                                transparent
                                onPress={() => this.segmentClicked(2)}
                                active={this.state.activeIndex == 2}
                            >
                                <Icon name='ios-bookmark'
                                      style={[this.state.activeIndex == 2 ? {} : {color: 'grey'}]}/>
                            </Button>
                        </View>

                        {this.renderSection()}
                    </View>
                </Content>
            </Container>
        )
    }
}

export default MeTab;

const styles = StyleSheet.create({
    container:{
        flex: 1,
        alignItems:'center',
        justifyContent:'center',
    }
})