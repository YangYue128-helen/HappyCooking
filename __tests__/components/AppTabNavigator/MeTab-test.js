import React from 'react';
import MeTab from "../../../src/components/AppTabNavigator/MeTab";

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<MeTab />).toJSON();
    expect(tree).toMatchSnapshot();
});