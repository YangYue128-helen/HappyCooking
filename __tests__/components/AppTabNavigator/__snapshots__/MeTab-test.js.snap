// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "white",
        "flex": 1,
      },
    ]
  }
>
  <View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "backgroundColor": "#F8F8F8",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F8F8F8",
            "borderBottomColor": "#a7a6ab",
            "borderBottomWidth": 0.5,
            "elevation": 3,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "center",
            "left": 0,
            "paddingLeft": 6,
            "paddingRight": 10,
            "paddingTop": 18,
            "right": 0,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            }
          }
        >
          <Text />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flex": 1,
            }
          }
        >
          <Text>
            HelenYang
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }
        >
          <Text />
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={
        Array [
          Object {
            "padding": undefined,
          },
          undefined,
        ]
      }
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutomaticScroll={true}
      enableOnAndroid={false}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      resetKeyboardSpace={[Function]}
      resetScrollToCoords={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      scrollEventThrottle={1}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollIntoView={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
      update={[Function]}
      viewIsInsideTabBar={false}
    >
      <View>
        <View
          style={
            Object {
              "paddingTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "borderRadius": 37.5,
                    "height": 75,
                    "width": 75,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 3,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-around",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                    }
                  }
                >
                  <Text>
                    20
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "grey",
                        "fontSize": 10,
                      }
                    }
                  >
                    posts
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                    }
                  }
                >
                  <Text>
                    30
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "grey",
                        "fontSize": 10,
                      }
                    }
                  >
                    recipes
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                    }
                  }
                >
                  <Text>
                    50
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "grey",
                        "fontSize": 10,
                      }
                    }
                  >
                    followers
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "paddingTop": 10,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={false}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 1,
                      "borderColor": "#000",
                      "borderLeftWidth": 1,
                      "borderRadius": 5,
                      "borderRightWidth": 1,
                      "borderTopWidth": 1,
                      "borderWidth": 1,
                      "elevation": null,
                      "flex": 3,
                      "flexDirection": "row",
                      "height": 30,
                      "justifyContent": "center",
                      "marginLeft": 10,
                      "opacity": 1,
                      "paddingBottom": 6,
                      "paddingTop": 6,
                      "shadowColor": null,
                      "shadowOffset": null,
                      "shadowOpacity": null,
                      "shadowRadius": null,
                    }
                  }
                >
                  <Text>
                    Edit Profile
                  </Text>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 1,
                      "borderColor": "#000",
                      "borderLeftWidth": 1,
                      "borderRadius": 5,
                      "borderRightWidth": 1,
                      "borderTopWidth": 1,
                      "borderWidth": 1,
                      "elevation": null,
                      "flex": 1,
                      "flexDirection": "row",
                      "height": 30,
                      "justifyContent": "center",
                      "marginLeft": 5,
                      "marginRight": 10,
                      "opacity": 1,
                      "paddingBottom": 6,
                      "paddingTop": 6,
                      "shadowColor": null,
                      "shadowOffset": null,
                      "shadowOpacity": null,
                      "shadowRadius": null,
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 10,
                "paddingVertical": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Yang Yue
            </Text>
            <Text>
              Chinese cuisine | Korean cuisine
            </Text>
          </View>
        </View>
        <View>
          <View
            style={
              Object {
                "borderTopColor": "#eeeeee",
                "borderTopWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 5,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "borderWidth": undefined,
                  "elevation": 0,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "space-between",
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "shadowColor": null,
                  "shadowOffset": null,
                  "shadowOpacity": null,
                  "shadowRadius": null,
                }
              }
            >
              <Text />
            </View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 5,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "borderWidth": undefined,
                  "elevation": 0,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "space-between",
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "shadowColor": null,
                  "shadowOffset": null,
                  "shadowOpacity": null,
                  "shadowRadius": null,
                }
              }
            >
              <Text />
            </View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 5,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "borderWidth": undefined,
                  "elevation": 0,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "space-between",
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "shadowColor": null,
                  "shadowOffset": null,
                  "shadowOpacity": null,
                  "shadowRadius": null,
                }
              }
            >
              <Text />
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "width": 250,
                  },
                  Object {
                    "height": 250,
                  },
                  Object {
                    "marginBottom": 2,
                  },
                  Object {
                    "paddingRight": 0,
                  },
                ]
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "width": 250,
                  },
                  Object {
                    "height": 250,
                  },
                  Object {
                    "marginBottom": 2,
                  },
                  Object {
                    "paddingLeft": 2,
                  },
                ]
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "width": 250,
                  },
                  Object {
                    "height": 250,
                  },
                  Object {
                    "marginBottom": 2,
                  },
                  Object {
                    "paddingLeft": 2,
                  },
                ]
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "width": 250,
                  },
                  Object {
                    "height": 250,
                  },
                  Object {
                    "marginBottom": 2,
                  },
                  Object {
                    "paddingRight": 0,
                  },
                ]
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "width": 250,
                  },
                  Object {
                    "height": 250,
                  },
                  Object {
                    "marginBottom": 2,
                  },
                  Object {
                    "paddingLeft": 2,
                  },
                ]
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "width": 250,
                  },
                  Object {
                    "height": 250,
                  },
                  Object {
                    "marginBottom": 2,
                  },
                  Object {
                    "paddingLeft": 2,
                  },
                ]
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "width": 250,
                  },
                  Object {
                    "height": 250,
                  },
                  Object {
                    "marginBottom": 2,
                  },
                  Object {
                    "paddingRight": 0,
                  },
                ]
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "width": 250,
                  },
                  Object {
                    "height": 250,
                  },
                  Object {
                    "marginBottom": 2,
                  },
                  Object {
                    "paddingLeft": 2,
                  },
                ]
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "width": 250,
                  },
                  Object {
                    "height": 250,
                  },
                  Object {
                    "marginBottom": 2,
                  },
                  Object {
                    "paddingLeft": 2,
                  },
                ]
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "width": 250,
                  },
                  Object {
                    "height": 250,
                  },
                  Object {
                    "marginBottom": 2,
                  },
                  Object {
                    "paddingRight": 0,
                  },
                ]
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
